File: funboxFrontend/WebContent/controller/friendcontroller.js
/**
 * FriendController
 */
app.controller('FriendCtrl',function($scope,$location,FriendService,$rootScope){
	function getSuggestedUsers(){
		FriendService.getSuggestedUsers().then(function(response){
			//response.data? List<User> - suggestedUsers
		$scope.suggestedUsers=response.data
		},function(response){
			$scope.error=response.data
			if(response.status==401)
				$location.path('/login')
		
		})
		
		
	}
	 getSuggestedUsers()
	 
	 $scope.sendFriendRequest=function(toIdValue){
		 FriendService.sendFriendRequest(toIdValue).then(function(response){
			 alert('Friend request has been sent successfully..')
			 getSuggestedUsers()
		 },function(response){
			 $scope.error=response.data
				if(response.status==401)
					$location.path('/login')
		 })
	 }
	 function getPendingRequests(){
		 FriendService.getPendingRequests().then(function(response){
			 //response.data-> Type of data List<Friend>
			 $scope.pendingRequests=response.data
		 },function(response){
			 $scope.error=response.data
				if(response.status==401)
					$location.path('/login')
		 
		 })
	 }
	  getPendingRequests()
	  $scope.updateStatus=function(friendRequest,updatedStatus){
		//char updatedStatus=[A or D]
		  //friendRequest is Friend object
		  //friendRequest.status=A or friendRequest.status=D
		  friendRequest.status=updatedStatus; //[A or D]Friend {id=41,status='A',fromId,toId}
		  FriendService.updateStatus(friendRequest).then(
				  function(response){
			     getPendingRequests()
		  },
		  function(response){
			  $scope.error=response.data
				if(response.status==401)
					$location.path('/login')
		  })
	  }
	      function getAllFriends(){
	    	  FriendService.getAllFriends().then(function(response){
	    		//response.data=?
	    		  $scope.friends=response.data
	    	  },function(response){
	    		  $scope.error=response.data
					if(response.status==401)
						$location.path('/login')
	    	  
	    	  })
	      }
	       getAllFriends()
})

File: funboxMiddleware/src/main/java/com/text/controller/FriendController.java
package com.text.controller;

import java.util.List;

import javax.servlet.http.HttpSession;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import com.text.Dao.FriendDao;
import com.text.Dao.UserDao;
import com.text.model.ErrorClazz;
import com.text.model.Friend;
import com.text.model.User;


@RestController
public class FriendController {
	@Autowired
	private FriendDao friendDao;
	@Autowired
	private UserDao userDao;
	
@RequestMapping(value="/suggestedusers",method=RequestMethod.GET)
public ResponseEntity<?> getSuggestedUsers(HttpSession session){
	String email=(String)session.getAttribute("email"); 
	if(email==null){
	  ErrorClazz errorClazz=new ErrorClazz(7,"Unauthorized access..please login");
	  return new ResponseEntity<ErrorClazz>(errorClazz,HttpStatus.UNAUTHORIZED);
}
	
	List<User>suggestedUsers=friendDao.getSuggestedUsers(email);
return new ResponseEntity<List<User>>(suggestedUsers,HttpStatus.OK);
}


@RequestMapping(value="/addfriend",method=RequestMethod.POST)
public ResponseEntity<?> addFriend(@RequestBody User toIdValue,HttpSession session){
	String email=(String)session.getAttribute("email"); 
	if(email==null){
		ErrorClazz errorClazz=new ErrorClazz(7,"Unauthorized access..please login");
	return new ResponseEntity<ErrorClazz>(errorClazz,HttpStatus.UNAUTHORIZED);
	}
	User fromId=userDao.getUser(email);
	Friend friend=new Friend();
	friend.setFromId(fromId);
	friend.setToId(toIdValue);
	friend.setStatus('P');
	friendDao.addFriend(friend);
	
	return new ResponseEntity<Void>(HttpStatus.OK);
}

@RequestMapping(value="/pendingrequests",method=RequestMethod.GET)
public ResponseEntity<?> getPendingRequests(HttpSession session){
	String email=(String)session.getAttribute("email"); 
	if(email==null){
	  ErrorClazz errorClazz=new ErrorClazz(7,"Unauthorized access..please login");
	  return new ResponseEntity<ErrorClazz>(errorClazz,HttpStatus.UNAUTHORIZED);
}
	
	List<Friend> pendingRequests=friendDao.pendingRequests(email);
	return new ResponseEntity<List<Friend>>(pendingRequests,HttpStatus.OK);
	//List<Friend> -> pendingRequests
}


@RequestMapping(value="/updatestatus",method=RequestMethod.PUT)
public ResponseEntity<?> updateStatus(@RequestBody Friend friendRequest,HttpSession session){
	String email=(String)session.getAttribute("email"); 
	if(email==null){
	  ErrorClazz errorClazz=new ErrorClazz(7,"Unauthorized access..please login");
	  return new ResponseEntity<ErrorClazz>(errorClazz,HttpStatus.UNAUTHORIZED);
}
   friendDao.upadateStatus(friendRequest);
   return new ResponseEntity<Void>(HttpStatus.OK);
 }


@RequestMapping(value="/friends",method=RequestMethod.GET)
public ResponseEntity<?> getAllFriends(HttpSession session){
	String email=(String)session.getAttribute("email"); 
	if(email==null){
	  ErrorClazz errorClazz=new ErrorClazz(7,"Unauthorized access..please login");
	  return new ResponseEntity<ErrorClazz>(errorClazz,HttpStatus.UNAUTHORIZED);
}
        List<Friend> friends=friendDao.getAllFriends(email);
        return new ResponseEntity<List<Friend>>(friends,HttpStatus.OK);
	
}
}